src/generic/datavgen.cpp

```diff
@@ -2686,14 +2846,25 @@ bool wxDataViewMainWindow::ItemDeleted(const wxDataViewItem& parent,
 bool wxDataViewMainWindow::ItemChanged(const wxDataViewItem & item)
 {
     SortPrepare();
-    GetModel()->Resort();
+
+    // Instead of resorting the parent, just move this specific child node.
+    wxDataViewItem parentItem = GetModel()->GetParent(item);
+    wxDataViewTreeNode * parentNode = FindNode(parentItem);
+    if ( parentNode )
+    {
+        wxDataViewTreeNode * node = FindNode(item);
+        wxASSERT(node);
+        parentNode->MoveUpdatedChild(node);
+    }
 
     GetOwner()->InvalidateColBestWidths();
 
     // Send event
     wxDataViewEvent le(wxEVT_DATAVIEW_ITEM_VALUE_CHANGED, m_owner, item);
     m_owner->ProcessWindowEvent(le);
 
+	// Make sure the change is actually shown right away
+	UpdateDisplay();
```

Comment:
My apologies, I confused it with `RecalculateDisplay()`, mostly because I *knew* that every change causes idle-time repaint — that was a side-effect of always calling `Resort()`, though, so I think that got broken by your changes here. IOW, moving the `UpdateDisplay` call to be explicit here is the right thing to do, sorry for the noise.